frontend/src/hooks/useSearchParamUpdater.ts

```diff
@@ -0,0 +1,59 @@
+import { usePathname, useSearchParams } from "next/navigation";
+
+export function useSearchParamUpdater() {
+  const searchParams = useSearchParams();
+  const pathname = usePathname() || "";
+
+  const updateSingularParam = (value: string, key: string) => {
+    const params = new URLSearchParams(searchParams || {});
+
+    if (value) {
+      params.set(key, "QUERY_PLACEHOLDER");
+    } else {
+      params.delete(key);
+    }
+
+    let newPath = `${pathname}?${params.toString()}`;
+    newPath = newPath.replace("QUERY_PLACEHOLDER", value);
+
+    // TODO - expand this to other filters as they are built,
+    // such as agency and funding instrument,
+    // so we retain commas instead of %2C in the URL
+    if (params.get("status")) {
+      const statusCheckboxParams = params.get("status") || "";
+      const encodedStatusCheckboxParams = statusCheckboxParams.replaceAll(
+        ",",
+        "%2C",
+      );
```

Comment:
Doesn't this replace the commas with `%2C`?

Reply:
When we use the Next `useSearchParams` hook, it's toString function gives `%2C` 

In order to replace that back to commas, I was using a reference string to do the replace:
```
  newPath = newPath.replace(
        encodedStatusCheckboxParams,
        statusCheckboxParams,
      );
```

I'm going to try to clean this code up a bit - it's a bit convoluted